### ISSUE TITLE
ContentBasedDeduplication not working correctly in FIFO queue

### ISSUE THREAD

**Issue by gkpty** [localstack/localstack#6327]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I created an SQS FIFO queue with content based deduplication and sent a message that contains the same messageBody and attributes twice.

The message gets sent and received by the queue twice.

Im using the latest version of localstack with docker-compose.

### Expected Behavior

When ContetBasedDeduplication is set to true a message with the same body should only be recieved once by the queue during the 5 minute deduplication interval.

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

create queue
```
sqs_client.create_queue(
            QueueName='my_queue.fifo',
            Attributes={
                'FifoQueue': 'true',
                'ContentBasedDeduplication': 'true'
            },
        )
```

send message
```
item = {'hello': 'hola'}
sqs_client.send_message(
            QueueUrl='my_queue.fifo',
            MessageBody=json.dumps(item),
            MessageAttributes= {'hello': {'StringValue': 'hola', 'DataType': 'String'}},
            MessageGroupId='group-1'
        )
```

### Environment

```markdown
- OS:Ubuntu 20.04
- LocalStack: latest
```

**Comment by bwiercinski** [comment]
in addition this bug only occurs when `--message-deduplication-id` parameter is not passed while sending a message. 

this is absolutely fine according to the AWS documentation: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html:
> If you aren't able to provide a MessageDeduplicationId and you enable ContentBasedDeduplication for your queue, Amazon SQS uses a SHA-256 hash to generate the MessageDeduplicationId using the body of the message (but not the attributes of the message).

summary:
```sh
awslocal sqs create-queue --queue-name input.fifo --attributes '{"FifoQueue": "true", "FifoThroughputLimit": "perMessageGroupId", "ContentBasedDeduplication": "true", "DeduplicationScope": "messageGroup"}'
awslocal sqs send-message --queue-url http://localhost:4566/000000000000/input.fifo --message-body A --message-group-id G1
awslocal sqs send-message --queue-url http://localhost:4566/000000000000/input.fifo --message-body A --message-group-id G1
awslocal sqs receive-message --max-number-of-messages 10 --visibility-timeout 2 --queue-url http://localhost:4566/000000000000/input.fifo # 2 messages instead of 1
```

using this, will work:
```sh
awslocal sqs send-message --queue-url http://localhost:4566/000000000000/input.fifo --message-body A --message-group-id G1 --message-deduplication-id A
```

**Comment by thrau** [comment]
hey @bwiercinski , thanks for the extra hint, that was very helpful. it's fixed now and will be in the latest image in about an hour

**Comment by bwiercinski** [comment]
no problem, thanks :)

**Comment by bwiercinski** [comment]
@thrau  when it will be released as a tagged version eg 1.0.1 ?
